### PR TITLE
Block match

### PR DESCRIPTION
**Pull request by dmcassel**

### CODE REVIEW

File: src/main/ml-modules/ext/com.marklogic.smart-mastering/matcher.xqy [hunk 1]
```diff
@@ -135,7 +140,11 @@ declare function matcher:find-document-matches-by-options(
         else (),
         cts:or-query(
           $minimum-threshold-combinations
-        )
+        ),
+        let $blocks := matcher:get-blocks(fn:base-uri($document))
+        where $blocks
```
Comment (paxtonhare) [inline src/main/ml-modules/ext/com.marklogic.smart-mastering/matcher.xqy]: will this always fn:exist()? seems like the code below is returning an array always.

File: src/main/ml-modules/ext/com.marklogic.smart-mastering/matcher.xqy [hunk 2]
```diff
@@ -522,3 +531,67 @@ declare function matcher:lock-on-search($query-results)
...
+    sem:sparql(
+      "select distinct(?uri as ?blocked) where { ?uri ?isBlocked ?target }",
+      map:new((
+        map:entry("target", sem:iri($uri)),
+        map:entry("isBlocked", $PRED-MATCH-BLOCK)
+      )),
+      "map"
+    )
+  return
+    array-node {
```
Comment (paxtonhare) [inline src/main/ml-modules/ext/com.marklogic.smart-mastering/matcher.xqy]: should this be...?

```
if (fn:exists($solution)) then
  array-node { map:get($solution, "blocked") }
else ()
```

File: src/test/ml-modules/root/test/suites/matching/allow-match.xqy [hunk 3]
```diff
@@ -0,0 +1,31 @@
+xquery version "1.0-ml";
+
+import module namespace matcher = "http://marklogic.com/smart-mastering/matcher"
+at "/ext/com.marklogic.smart-mastering/matcher.xqy";
+
+import module namespace test = "http://marklogic.com/roxy/test-helper" at "/test/test-helper.xqy";
+
+(: Force update mode :)
```
Comment (paxtonhare) [inline src/test/ml-modules/root/test/suites/matching/allow-match.xqy]: can you add disable function mapping to these? just for fun?

File: src/main/ml-modules/ext/com.marklogic.smart-mastering/matcher.xqy (hunk 1 above)
Comment (dmcassel) [reply src/main/ml-modules/ext/com.marklogic.smart-mastering/matcher.xqy]: you're right. should be checking fn:exists($blocks/node())

File: src/test/ml-modules/root/test/suites/matching/allow-match.xqy (hunk 3 above)
Comment (dmcassel) [reply src/test/ml-modules/root/test/suites/matching/allow-match.xqy]: oversight; will do

**Pull request merged**